### PR TITLE
module: refactor gateway script to unify checkers 

### PR DESCRIPTION
**Pull request by leemgs**
Fixed issue #299.

This commit is to refator a gateway script to unify the existing
two checkers such as format and audit.

**Changes proposed in this PR:**

* Version 2:
1. Refactor the gateway script after merging two check scripts
2. Unified the report folder consistently
3. Added annotations for maintenance

* Version 1:
1. Unified start-up code of the existing two checker
2. Removed duplicated statements
3. Merged two folders to one folder, pr-checker

Signed-off-by: Geunsik Lim <leemgs@gmail.com>

### CODE REVIEW

**Comment by leemgs** [comment]
The aging test is completed.

File: ci/taos/checker-pr-gateway.sh [hunk 1]
```diff
@@ -31,27 +31,99 @@
...
+
+echo -e "[DEBUG] current path: $(pwd)."                              | tee -a $log_file
+echo -e "[DEBUG] ./checker-pr-audit-async.sh $1 $2 $3 $4 $5 $6 "     | tee -a $log_file
+echo -e "[DEBUG] Starting asynchronouly...               "           | tee -a $log_file
+cd ./taos/
+./checker-pr-format-async.sh $1 $2 $3 $4 $5 $6                       | tee -a $log_file  &
+echo -e "[DEBUG] Running"                                            | tee -a $log_file
+echo -e "[DEBUG] ......"                                             | tee -a $log_file
+echo -e "[DEBUG] Completed"                                          | tee -a $log_file
+# --------------------------- Run module: audit ---------------------------------------------------------------
```
Comment (ohsewon) [inline ci/taos/checker-pr-gateway.sh]: Does run audit module after finishing format module?

Comment (leemgs) [reply ci/taos/checker-pr-gateway.sh]: Updated.

File: ci/taos/checker-pr-gateway.sh [hunk 2]
```diff
@@ -31,27 +31,99 @@
...
+# Check if dir_commit folder exists, then, create dir_commit folder
+# Let's keep the existing result although the same target directory already exists.
+cd $dir_ci
+dir_worker="repo-workers/pr-checker"
+export dir_worker=$dir_worker
+export dir_commit=${dir_worker}/${input_date}-${input_pr}-${input_commit}
```
Comment (ohsewon) [inline ci/taos/checker-pr-gateway.sh]: 100~105 lines are duplicated with 69~74

File: ci/taos/checker-pr-gateway.sh [hunk 3]
```diff
@@ -31,27 +31,99 @@
...
+
+# Remove existing folders
+if [[ -d $dir_commit ]]; then
+    echo "[DEBUG] WARN: mkdir command is failed because $dir_commit directory already exists."
+    echo "[DEBUG] WARN: So removing the existing directory..."
+    rm -rf ./${dir_commit}
+fi
```
Comment (ohsewon) [inline ci/taos/checker-pr-gateway.sh]: 107~112 need to be removed.

File: ci/taos/checker-pr-gateway.sh [hunk 4]
```diff
@@ -31,27 +31,99 @@
...
+
+
+# Save a log file for debugging
+mkdir -p $dir_commit
```
Comment (ohsewon) [inline ci/taos/checker-pr-gateway.sh]: this line is unnecessary

File: ci/taos/checker-pr-gateway.sh [hunk 5]
```diff
@@ -31,27 +31,99 @@
...
+log_file="${dir_ci}/${dir_commit}/checker-pr-audit.log"
+echo -e "[DEBUG] dir_commit is $dir_commit. This folder is created." | tee    $log_file
+echo -e "[DEBUG] Initializing...                         "           | tee -a $log_file
+echo -e "[DEBUG] ./checker-pr-audit.sh $1 $2 $3 $4 $5 $6 "           | tee -a $log_file
+
+echo -e "[DEBUG] current path: $(pwd)."                              | tee -a $log_file
+echo -e "[DEBUG] ./checker-pr-audit-async.sh $1 $2 $3 $4 $5 $6 "     | tee -a $log_file
+echo -e "[DEBUG] Starting asynchronouly...               "           | tee -a $log_file
+cd ./taos/
```
Comment (ohsewon) [inline ci/taos/checker-pr-gateway.sh]: IMO, this line will make error

**Comment by leemgs** [comment]
**Changes proposed in this PR:**

* Version 2:
1. Refactor the gateway script after merging two check scripts
2. Unified the report folder consistently

* Version 1:
1. Unified start-up code of the existing two checker
2. Removed duplicated statements
3. Merged two folders to one folder, pr-checker

**Comment by leemgs** [comment]
**Changes proposed in this PR:**

* Version 2:
1. Refactor the gateway script after merging two check scripts
2. Unified the report folder consistently
3. Added annotations for maintenance

* Version 1:
1. Unified start-up code of the existing two checker
2. Removed duplicated statements
3. Merged two folders to one folder, pr-checker

**Pull request closed without merging**